VideoLocker/src/main/java/org/edx/mobile/module/analytics/ISegmentImpl.java

```diff
@@ -89,6 +87,80 @@ public Traits identifyUser(String userID, String email, String username) {
         return traits;
     }
 
+    /**
+     * This function is used to send the screen tracking event.
+     *
+     * @param screenName The screen name to track
+     * @return A {@link Properties} object populated with analytics-event info
+     */
+    @Override
+    public Properties trackScreenView(@NonNull String screenName) {
+        return trackScreenView(screenName, null, null);
+    }
+
+    /**
+     * This function is used to send the screen tracking event, with an extra event for
+     * sending course id.
+     *
+     * @param screenName The screen name to track
+     * @param courseId   course id of the course we are viewing
+     * @param action     any custom action we need to send with event
+     * @return A {@link Properties} object populated with analytics-event info
+     */
+    @Override
+    public Properties trackScreenView(@NonNull String screenName, @Nullable String courseId,
+                                      @Nullable String action) {
```

Comment:
There's a catch here.

Previously we were sending the `courseId` inside the `properties` dictionary. According to the new requirements it should now go inside the `data` dictionary.

++ @symbolist